padl/transforms.py

```diff
@@ -190,6 +190,8 @@ def _pd_merge_components(components):
             return Transform._pd_merge_components(res)
         return res
 
+    @property
+    @lru_cache(maxsize=128)
```

Comment:
@wuhu Since this is cached now do we need to store it into `self._pd_stages`? Could we do something like this 
```python
    @property
    @lru_cache(maxsize=128)
    def _pd_stages(self):
        _, splits, has_batchify = self._pd_splits()
        if has_batchify:
            preprocess, forward, postprocess = splits
        else:
            preprocess, forward, postprocess = identity, splits[0], splits[2]
        return preprocess, forward, postprocess
```

Reply:
yes! makes sense